src/predict_request_validation_utils.cpp

```diff
@@ -0,0 +1,301 @@
+    uint64      data in request.tensor_content
+    float16     request.tensor_content is empty, data located in request.half_val
+    float32     data in request.tensor_content
+    double      data in request.tensor_content
+
+    _TENSOR_CONTENT_TYPES
+    https://github.com/tensorflow/tensorflow/blob/903a6399aab19b549fefd0ead836af644f3d00f8/tensorflow/python/framework/tensor_util.py#L237
+*/
+
+    size_t expectedValueCount = 1;
+    for (int i = 0; i < proto.tensor_shape().dim_size(); i++) {
+        expectedValueCount *= proto.tensor_shape().dim(i).size();
+    }
+
+    // Network expects tensor content size or value count
+    if (proto.dtype() == tensorflow::DataType::DT_UINT16) {
+        if (proto.int_val_size() < 0 ||
+            expectedValueCount != static_cast<size_t>(proto.int_val_size())) {
+            std::stringstream ss;
+            ss << "Expected: " << expectedValueCount << "; Actual: " << proto.int_val_size();
+            const std::string details = ss.str();
+            SPDLOG_DEBUG("Invalid number of values in tensor proto container - {}", details);
+            return Status(StatusCode::INVALID_VALUE_COUNT, details);
+        }
+    } else if (proto.dtype() == tensorflow::DataType::DT_HALF) {
+        if (proto.half_val_size() < 0 ||
+            expectedValueCount != static_cast<size_t>(proto.half_val_size())) {
+            std::stringstream ss;
+            ss << "Expected: " << expectedValueCount << "; Actual: " << proto.half_val_size();
+            const std::string details = ss.str();
+            SPDLOG_DEBUG("Invalid number of values in tensor proto container - {}", details);
+            return Status(StatusCode::INVALID_VALUE_COUNT, details);
+        }
+    } else {
+        size_t expectedContentSize = expectedValueCount * expectedPrecision.size();
+        if (expectedContentSize != proto.tensor_content().size()) {
+            std::stringstream ss;
+            ss << "Expected: " << expectedContentSize << " bytes; Actual: " << proto.tensor_content().size() << " bytes";
+            const std::string details = ss.str();
+            SPDLOG_DEBUG("Invalid content size of tensor proto - {}", details);
+            return Status(StatusCode::INVALID_CONTENT_SIZE, details);
+        }
+    }
+    return StatusCode::OK;
+}
+
+Status validateNumberOfShapeDimensions(const ovms::TensorInfo& inputInfo, const tensorflow::TensorProto& proto) {
+    // Network and request must have the same number of shape dimensions, higher than 0
+    const auto& shape = inputInfo.getEffectiveShape();
+    if (proto.tensor_shape().dim_size() <= 0 ||
+        shape.size() != static_cast<size_t>(proto.tensor_shape().dim_size())) {
+        std::stringstream ss;
+        ss << "Expected: " << TensorInfo::shapeToString(shape)
+           << "; Actual: " << TensorInfo::tensorShapeToString(proto.tensor_shape());
+        const std::string details = ss.str();
+        SPDLOG_DEBUG("Invalid number of shape dimensions - {}", details);
```

Comment:
Use one convention  - ": " instead of " - "  in logs.

Reply:
`-` convention is specific for splitting spdlog from clientside message.
This is consistent since separate messages use `:` and are splitted with `-`